xl2times/transforms.py

```diff
@@ -811,20 +814,45 @@ def process_regions(
         else:
             print("WARNING: Regions filter not applied; no valid entries found. ")
 
+    return tables
+
+
+def complete_dictionary(
+    config: datatypes.Config,
+    tables: Dict[str, DataFrame],
+    model: datatypes.TimesModel,
+) -> Dict[str, DataFrame]:
+
     for k, v in {
         "AllRegions": model.all_regions,
         "Regions": model.internal_regions,
+        "DataYears": model.data_years,
+        "PastYears": model.past_years,
+        "ModelYears": model.model_years,
     }.items():
-        tables.append(
-            datatypes.EmbeddedXlTable(
-                tag=k,
-                uc_sets={},
-                sheetname="",
-                range="",
-                filename="",
-                dataframe=pd.DataFrame(v, columns=["region"]),
-            )
-        )
+        if "region" in k.lower():
+            column_list = ["region"]
+        else:
+            column_list = ["year"]
+
+        tables[k] = pd.DataFrame(v, columns=column_list)
+
+    # Dataframes
+    for k, v in {
+        "Attributes": model.attributes,
+        "Commodities": model.commodities,
+        "CommodityGroups": model.commodity_groups,
+        "CommodityGroupMap": model.com_gmap,
+        "Processes": model.processes,
+        "Topology": model.topology,
+        "Trade": model.trade,
+        "TimePeriods": model.time_periods,
+        "TimeSlices": model.ts_tslvl,
+        "TimeSliceMap": model.ts_map,
+        "UserConstraints": model.user_constraints,
+        "Units": model.units,
+    }.items():
+        tables[k] = v
```

Comment:
Is this intended to be a temporary solution -- that we copy the tables back from the `TimesModel` instance to the `Dict[str, DataFrame]` so that the `produce_times_tables` can work as before? I'm wondering if we'd eventually want to move to encapsulating all the model data in `TimesModel` and having a `TimesModel.produce_tables` function that produces the output.

Reply:
Yes, that's right. Switching to e.g. `TimesModel.to_gams()` later on would be great!